### ISSUE TITLE
[unbound-method] False positive on `Intl.Collator.compare` (reopen #4383)

### ISSUE THREAD

**Issue by jcranendonk** [typescript-eslint/typescript-eslint#7098]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) ~~and found none that matched my issue~~ (see description)
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

[Minimal example](https://typescript-eslint.io/play/#ts=4.5.2&sourceType=module&code=PQKgBApgzgNglgOwC5gEQAEkE8AO0DGATnDkgLTTzLACuCARgPZ0AmZAthEgBaMuoAuNBEKFGhVGBDAAsACh5+RgigolMGAEMk4sAF4wCCAHcwASWQwAdAGFGG7eIAUqCAlQBKRctVgo4pAgWfTAAbVQAM0ZGVAAaNHpNCQBdK39CJCd1LR1CKyV2HCSILzkgA&rules=N4XyA&tsConfig=N4XyA)

### Repro Code

```typescript
const collator = new Intl.Collator("en")
const sorted = ["foo", "bar"].sort(collator.compare)
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/unbound-method": "error"
  },
};
```

### Expected Result

The rule should not fire an error for this code, as `Intl.Collator.prototype.compare` is [bound to _collator_ according to the ECMAScript spec](https://tc39.es/ecma402/#sec-intl.collator.prototype.compare).

### Actual Result

The rule fires for the use of `collator.compare`:

```
Avoid referencing unbound methods which may cause unintentional scoping of `this`.
If your function does not access `this`, you can annotate it with `this: void`, or consider using an arrow function instead. 4:36 - 4:52
```

### Additional Info

This is a duplicate of #4383, which was marked 'unable to repro'. However, the playground example linked in the team response actually reproduces the issue.

**Comment by Josh-Cena** [comment]
@JoshuaKGoldberg That one's irrelevant; that one's about accepting `Intl.Locale` objects in place of locale identifiers. To fix the lib, we would need to do something like:

```ts
interface Collator {
  get compare(): (x: string, y: string) => number;
}
```

**Comment by JoshuaKGoldberg** [comment]
Cool - either way, it's a TypeScript thing and not us. Cheers!

**Comment by bradzacher** [comment]
@JoshuaKGoldberg we do have a code which lists and resolves native JS API methods that are static and known to be bound ahead of time:

https://github.com/typescript-eslint/typescript-eslint/blob/a91bb9e264544fc635a44468c2543b94fa83ac1c/packages/eslint-plugin/src/rules/unbound-method.ts#L47-L100

We did this because the TS types are wrong and likely won't ever get fixed (based on our past experience with things like `no-base-to-string`).

However we don't have any config for ***instance*** methods. Perhaps we should consider adding config for instance methods?

**Comment by thw0rted** [comment]
Like Brad, I don't expect them to actually fix it, but I put in an issue with TS anyway.

**Comment by thw0rted** [comment]
Attn @JoshuaKGoldberg / @bradzacher , in case you're curious there is an [open MR to fix this on the TS side](https://github.com/microsoft/TypeScript/pull/62052) , but it will be as a method with `this: void` argument rather than a getter.  Probably OK?  I'm not sure how to test whether the proposed changes will close this issue or not so I thought it might be worth having someone from your team weigh in before the code is merged.

ETA: sorry, I didn't know about the ESLint Playground -- I was able to put [this repro](https://typescript-eslint.io/play/#code=PQKgBApgzgNglgOwC5gEQAEkE8AO0DGATnDkgLTTzLACuCARgPZ0AmZAthEgBaMuoAuNBEKFGhVGBDAAsACh5AbVQAzRo1QAaNPQCGEgLoA6KOKQAKBBADuYAJLIYRgMKMYMXUnHmA5BAQ%2BAJRG%2BIzsOPoQgfLy%2BB5QUGCu7p7icFDcYADe8mBgoeGR5jzpAgBujHAs2rpCUEjECADm2vQC9Y1NgdlghFw0hAhgukYwjPi6MBCuhX3m9N0AvvLLcspqGtqoeoYmZpY2SW4eXsQZ5sEFEX3RckA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYwAviNUA&tokens=false) together which means the MR is probably good?

**Comment by JoshuaKGoldberg** [comment]
Yup, `this: void` is a good idea.
